packages/utils/src/createPromiseManager.ts

```diff
@@ -0,0 +1,106 @@
+import { createDeferred, Deferred } from './createDeferred';
+
+type ManagedPromise = Deferred<any, number> & { deadline: number };
```

Comment:
I'm again not sure what you mean. There is unfortunately no way how to correctly type every deferred promise inside the manager. Do you mean to type the manager as a whole with the union of every possible promise value? It could actually make sense.

Reply:
Like this? https://github.com/trezor/trezor-suite/pull/10787/commits/913f7bc2207c9c5c6e4c49437d1f1f1427c55a13

Reply:
I just got my lizard brain triggered by any without any specific idea what to do about it
this could be useful 👍